lib/util/http/retry-after.ts

```diff
@@ -0,0 +1,86 @@
+import { setTimeout } from 'timers/promises';
+import { RequestError } from 'got';
+import { logger } from '../../logger';
+import * as hostRules from '../host-rules';
+import { parseUrl } from '../url';
+import type { Task } from './types';
+
+const hostBlocks = new Map<string, Promise<void>>();
+
+const maxRetries = 2;
+
+export async function wrapWithRetry<T>(
+  url: string,
+  task: Task<T>,
+  getDelaySeconds: (err: unknown) => number | null,
+  maxRetryAfter = 60,
+): Promise<T> {
+  const key = parseUrl(url)?.host ?? /* istanbul ignore next */ url;
+
+  let retries = 0;
+  for (;;) {
+    try {
+      const delayPromise = hostBlocks.get(key);
+      if (delayPromise) {
+        await delayPromise;
+        hostBlocks.delete(key);
+      }
+
+      return await task();
+    } catch (err) {
+      const delaySeconds = getDelaySeconds(err);
+      if (delaySeconds === null) {
+        throw err;
+      }
+
+      if (retries === maxRetries) {
+        logger.debug({ url }, 'Retry-After: giving up');
+        throw err;
+      }
+
+      logger.debug(
+        { url, delaySeconds },
+        `Retry-After: will retry after ${delaySeconds} seconds`,
+      );
+
+      const delay = 1000 * Math.max(0, Math.min(delaySeconds, maxRetryAfter));
```

Comment:
I think there might be a misunderstanding of the usage of maxRetryAfter. If the retry-after value from the server is more than maxRetryAfter then we should log and abort, and not "ignore" it like above